pandas/_libs/tslibs/period.pyx

```diff
@@ -1255,6 +1255,33 @@ cdef class _Period(object):
 
     @property
     def dayofyear(self):
+        """
+        Return the day of the year.
```

Comment:
Periods represent time spans so it's good to note that this is the start of the period. So maybe

    Return the day of the year the of the Period's start.

That's a bit wordy though. Feel free to reword if you have a better phrasing.

Reply:
sorry, but I didn't understand the meaning of the sentence `Return the day of the year the of the Period's start.`, specifically of the `period's start`

Reply:
Period is a span, so saying "Day of the year" is a bit ambiguous. Is it the start, end, somewhere in between?

If you have a period like `Period('2017-01-01', freq='A')

```python
In [2]: pd.Period('2017-01-01', freq='A')
Out[2]: Period('2017', 'A-DEC')

In [3]: pd.Period('2017-01-01', freq='A').day
Out[3]: 31
```

I was apparently wrong about it being the start. I guess it's related to `freq`

Reply:
This comment will be relevant to many of the Period attributes like day, dayofyear, .. (also the ones I already merged ..). 
So it would be good to be consistent in it.

The annual frequency is by default a "year-end" frequency, so it's indeed freq related.

Reply:
I'm sorry, I don't follow... Can you show two cases creating a period starting on the same day but with different frequencies, and how this affect `.day`, `.dayofweek`, `.dayofyear`, etc.?

Reply:
@dukebody small example:

```
In [30]: p1 = pd.Period('2017-01-01', freq='D')

In [31]: p2 = pd.Period('2017-01-01', freq='M')

In [32]: p1
Out[32]: Period('2017-01-01', 'D')

In [33]: p2
Out[33]: Period('2017-01', 'M')

In [34]: p1.start_time
Out[34]: Timestamp('2017-01-01 00:00:00')

In [35]: p2.start_time
Out[35]: Timestamp('2017-01-01 00:00:00')

In [36]: p1.dayofyear
Out[36]: 1

In [37]: p2.dayofyear
Out[37]: 31
```

I am not really familiar with this frequency business, but you have certain frequences which are "anchored" at the beginning of the period, and others at the end of the period. But I don't directly see a way how to inspect this.

Reply:
That's quite confusing. Should we open a ticket to make `.dayofyear` etc. always refer to the start time of the period? Not to the "anchored" time. Otherwise these functions are very unpredictable IMO and therefore don't add a lot of value.

Reply:
Yes, I am writing one currently ;) 
I suppose those attributes will mainly be useful for frequencies <= 1D

Reply:
I'm OK with merging this PR without referring to the start or end date of the period, and then work on code to make this predictable and change all docstrings to refer to the period start date.